cpp/dolfinx/fem/utils.cpp

```diff
@@ -83,22 +83,28 @@ fem::create_element_dof_layout(const ufc_dofmap& dofmap,
   std::array<int, 4> num_entity_dofs;
   std::copy_n(dofmap.num_entity_dofs, 4, num_entity_dofs.data());
 
-  int dof_count = 0;
+  std::array<int, 4> num_entity_closure_dofs;
+  std::copy_n(dofmap.num_entity_closure_dofs, 4, num_entity_closure_dofs.data());
 
   // Fill entity dof indices
   const int tdim = mesh::cell_dim(cell_type);
   std::vector<std::vector<std::set<int>>> entity_dofs(tdim + 1);
+  std::vector<std::vector<std::set<int>>> entity_closure_dofs(tdim + 1);
   std::vector<int> work_array;
   for (int dim = 0; dim <= tdim; ++dim)
   {
     const int num_entities = mesh::cell_num_entities(cell_type, dim);
     entity_dofs[dim].resize(num_entities);
+    entity_closure_dofs[dim].resize(num_entities);
     for (int i = 0; i < num_entities; ++i)
     {
       work_array.resize(num_entity_dofs[dim]);
       dofmap.tabulate_entity_dofs(work_array.data(), dim, i);
       entity_dofs[dim][i] = std::set<int>(work_array.begin(), work_array.end());
-      dof_count += num_entity_dofs[dim];
+
+      work_array.resize(num_entity_closure_dofs[dim]);
```

Comment:
I wouldn't worry about using a work array here - the loop is small and only called once or twice in a computation. Fine to just allocate inside the loop.